releasing/releasing/go.mod

```diff
@@ -0,0 +1,5 @@
+module sigs.k8s.io/kustomize/releasing
+
+go 1.13
```

Comment:
I don't know the policy for this, but should this be go 1.14?

The best I could find is this incoming PR to k8s: https://github.com/kubernetes/kubernetes/pull/88638

Reply:
This particular tool can go to 1.14 right away, as its entirely standalone.

Other in-repo modules are at 1.13. 

1.14 came out feb 25, almost 3 months ago, so yeah, its time to upgrade.  

Was going to release kustomize  at 3.6.0 today leaving go at 1.13, then after that upgrade to go 1.14 for the next minor release of kustomize (3.7).  sound reasonable?